docs/source/setup.rst

```diff
@@ -0,0 +1,534 @@
+    --psk PSK             Pre-shared key
+    --timeout TIMEOUT     Metrics generation timeout in seconds (default: 300)
+    --verbose, -v         Verbose logging
+    --web-port WEB_PORT   Web Port (default 9955)
+    --web-host WEB_HOST   IP to bind to (default: ::)
+    --web-path WEB_PATH   Web root in browser (default: /)
+
+ENV Prefix: ``CKMETRICS_``  
+Every CLI arg can also be specified using ENV variables.
+
+For instance the boolean ``--verbose`` would become ``CKMETRICS_VERBOSE=true`` or ``--timeout 300`` would become ``CKMETRICS_TIMEOUT=300``.
+
+Once started :ref:`setup-ckmetrics` will register for ``generate_metrics`` core events. When such an event is received it will
+generate Cloudkeeper metrics and provide them at the ``/metrics`` endpoint.
+
+As mentioned in the :ref:`prometheus` setup your configuration needs to contain this configuration snippet.
+
+.. code-block:: yaml
+    :caption: :ref:`prometheus` configuration snippet
+
+    scrape_configs:
+    - job_name: "ckmetrics"
+        static_configs:
+        - targets: ["localhost:9955"]
+
+Run ckmetrics
+-------------
+Now you can connect :ref:`setup-ckmetrics` to the previous setup :ref:`setup-ckcore` as well as let your prometheus connect to :ref:`setup-ckmetrics`.
+Please match your parameter values to reflect your environment while running :ref:`setup-ckcore`.
+
+We add the ``--verbose`` on first start to get used to what is happening exactly.
+You can skip this argument later to reduce log output volume when all components are set up.
+
+Add ``--ckcore-uri`` and ``--ckcore-ws-uri`` if :ref:`setup-ckcore` is running on another instance or port.
+Add ``--ckcore-graph`` if you defined another name of the graph for :ref:`setup-ckworker`
+
+.. code-block:: bash
+    :caption: Run ckmetrics
+
+    $ ckmetrics --verbose \
+      --ckcore-uri ${CKCORE_URI:-http://127.0.0.1:8900} \
+      --ckcore-ws-uri ${CKCORE_WS_URI:-ws://127.0.0.1:8900} \
+      --ckcore-graph ${CKCORE_GRAPH:-ck}
+
+.. code-block:: console
+    :caption: Successful launch log output
+
+    2021-10-05 13:20:43,798 - DEBUG - 6143/MainThread - generating metrics
+    2021-10-05 13:20:43,798 - INFO - 6143/webserver - CherryPy ENGINE Bus STARTING
+    2021-10-05 13:20:43,798 - DEBUG - 6143/ckmetrics - Registering <bound method CoreActions.shutdown of <CoreActions(ckmetrics, started 6189232128)>> with event SHUTDOWN (blocking: False, one-shot: False)
+    2021-10-05 13:20:43,798 - INFO - 6143/ckmetrics - Connecting to ckcore message bus
+    [...]
+    2021-10-05 13:20:43,824 - DEBUG - 6143/ckmetrics - ckmetrics connected to ckcore message bus
+    2021-10-05 13:20:44,904 - INFO - 6143/webserver - CherryPy ENGINE Serving on http://:::9955
+    2021-10-05 13:20:44,905 - INFO - 6143/webserver - CherryPy ENGINE Bus STARTED
+
+You can now access the metrics interface via `ckmetrics <http://localhost:9955/metrics>`_.
+
+.. _setup-cksh:
+
+cksh
+====
+
+Install cksh
```

Comment:
I would install `cksh` right after `ckcore` so that the user can already test a `> help` to see if the connection is working.